src/RuleTestCase.php

```diff
@@ -0,0 +1,148 @@
+<?php declare(strict_types = 1);
+
+namespace ShipMonk\PHPStanDev;
+
+use LogicException;
+use PHPStan\Analyser\Error;
+use PHPStan\Rules\Rule;
+use PHPStan\Testing\RuleTestCase as OriginalRuleTestCase;
+use function array_values;
+use function explode;
+use function file_get_contents;
+use function file_put_contents;
+use function implode;
+use function ksort;
+use function preg_match;
+use function preg_match_all;
+use function preg_replace;
+use function sprintf;
+use function trim;
+use function uniqid;
+
+/**
+ * @template TRule of Rule
+ * @extends OriginalRuleTestCase<TRule>
+ */
+abstract class RuleTestCase extends OriginalRuleTestCase
+{
+
+    protected function analyseFile(string $file, bool $autofix = false): void
```

Comment:
Lets support multiple files as in [DCD](https://github.com/shipmonk-rnd/dead-code-detector/blob/18a28eb9c3ab1ced99bbe722a6889f1180034c7c/tests/Rule/RuleTestCase.php#L34) so that we can utilize it even there.

Reply:
When single analysis of multiple files is needed. Crucial e.g. for trait tests, where definition filepaths get tricky. See usages in DCD, it was addded there for this exact reason.